Fix shield sample build issue for x_nucleo_iks01a2

galak opened the pull request:
samples/shields/x_nucleo_iks01a2/sample.shields.x_nucleo_iks01a2 fails to build on a number of boards because they lack `arduino_header` in their dts.  Make the sample depend on `arduino_header` and update various boards to say they support this.

galak committed:
samples: shields: x_nucleo_iks01a2: Depend on arduino_gpio

There are build issues with this sample now that it requires that the
board dts have an arduino_header node.  Add depends_on: arduino_gpio
to sample.yaml so we only build this if the board has that header.

Also remove a duplicate line for excluding the disco_l475_iot1 platform.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: Add arduino_gpio & and arduino_i2c to board supported

Update a number of boards that have arduino_gpio and arduino_i2c
support in their dts files to show that they support that in the
board.yaml file.  This allows coverage on several shield tests that
utilize the tags 'arduino_gpio' and 'arduino_i2c'.

Exlucde stm32mp157c_dk2 from some of the samples right now since the
connector on the board doesn't support A2/A3.  Also remove the duplicate
of exluding disco_l475_iot1.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

jfischer-phytec-iot commented on samples/shields/x_nucleo_iks01a2/sample.yaml:
```diff
@@ -2,8 +2,7 @@ sample:
     harness: shield
     tags: shield
-    depends_on: arduino_i2c
+    depends_on: arduino_i2c arduino_header
```
Is not one of them redundant? A board can not have arduino_i2c without arduino_header.

galak replied:
I think arduino_header might be poorly named.  This might be better as arduino_gpio.

galak replied:
Add a commit to rename the depend on to `arduino_gpio`

avisconti reviewed (changes requested):
what about arm/stm32mp157c_dk2?
It includes arduino_r3_connector.dtsi where arduino_header is defined, but it does not
specify it as supported in the yaml file. Isn't it?

erwango reviewed (changes requested):
As mentionned in comments, `arduino_gpio` would be a better fit

galak committed:
samples: shields: x_nucleo_iks01a3: rename arduino_header depend

Change arduino_header depend to arduino_gpio as that is more meaningful.
Its easy to mistake arduino_header to mean the board has an arduino
header in general.  What we want to depend on is that we have the gpio
connector in the dts that has 'arduino_header' as the node label.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
> what about arm/stm32mp157c_dk2?
> It includes arduino_r3_connector.dtsi where arduino_header is defined, but it does not
> specify it as supported in the yaml file. Isn't it?

Missed it, now fixed.

avisconti reviewed (approved):
LGTM now

erwango commented on samples/shields/x_nucleo_iks01a3/sensorhub/sample.yaml:
```diff
@@ -2,8 +2,7 @@ sample:
-    depends_on: arduino_i2c arduino_header
-    platform_exclude: disco_l475_iot1
+    depends_on: arduino_i2c arduino_gpio
+    platform_exclude: disco_l475_iot1 stm32mp157c_dk2
```
Thinking again about it, I think we could also add a stm32mp157c_dk2 dedicated overlay in the shield removing the pins that don't match, rather than excl
This would allow to keep using the board wit the shield (which totally works).
What do you think?

galak replied:
Agreed, but more work needs to be done for that, let's address that after we get this merged to fix the build issues.

See but #19078

erwango replied:
I tested successfully the following:
`boards/shields/x_nucleo_iks01a2/boards/stm32mp157c_dk2.overlay`
```
&arduino_i2c {
	lsm303agr-magn@1e {
		/delete-property/ irq-gpios;	/* A3 */
	};

	lsm303agr-accel@19 {
		/delete-property/ irq-gpios;	/* A3 */
	};
};
```

erwango reviewed (changes requested):
Proposing an alternative to stm32mp157c_dk2 exclusion from shield sanitycheck

galak commented:
@erwango Going to merge this and handle fixing up `stm32mp157c_dk2` as a post PR.
